### ISSUE TITLE
How to get Groups of User

### ISSUE THREAD

**Issue by gabrielemilan** [simpleidserver/SimpleIdServer#26]
Hello, nice project!

I have a question, maybe is my wrong configuration.

The User schema has groups attribute set readonly:
{
  "name": "groups",
  "type": "complex",
  "multiValued": true,
  "descriptions": null,
  "required": false,
  "caseExact": false,
  "mutability": "readonly",
  "returned": "default",
  "uniqueness": "none",
  "subAttributes": [
    {
      "name": "value",
      "type": "string",
      "multiValued": false,
      "descriptions": null,
      "required": false,
      "caseExact": false,
      "mutability": "readonly",
      "returned": "default",
      "uniqueness": "none"
    }
  ]
}

Than on the Group schema I have a members attribute.
To add a new user to a spcific group I can use patch to Group, but what about getting groups from Users endpoint?

If I understood correctly, the values of members in Group representation is the id of user, so when I get specific user I should get also groups, am I right?

If so, is this behaviour implemented?

Thanks a lot :-)

**Comment by simpleidserver** [comment]
Hi,

To assign a user to a group you can execute a PATCH instruction with the following content :

```
{
     "schemas":
       ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
     "Operations": [
       {
         "op":"remove","path":"members"
       },
       {
         "op":"add",
         "path":"members",
         "value":[
         {
           "display": "Babs Jensen",
           "$ref":
   "https://example.com/v2/Users/2819c223...413861904646",
           "value": "2819c223-7f76-453a-919d-413861904646"
         },
         {
           "display": "James Smith",
           "$ref":
   "https://example.com/v2/Users/08e1d05d...473d93df9210",
           "value": "08e1d05d-121c-4561-8b96-473d93df9210"
         }]
       }
     ]
   }
```

Unfortunately the **groups** property present in the JSON representation of a **user**, doesn't contain the list of groups.

This issue will be fixed in the next release ASAP.

**Comment by simpleidserver** [comment]
The issue has been fixed in the release 1.1.1.

When a user is retrieved then his groups are also returned.
